pkg/testing/integration/program.go

```diff
@@ -1325,6 +1361,67 @@ func (pt *ProgramTester) TestLifeCycleInitialize() error {
 		}
 	}
 
+	// Environments
+	for _, env := range pt.opts.CreateEnvironments {
+		name := pt.opts.getEnvNameWithOwner(env.Name)
+
+		initArgs := []string{"env", "init", name}
+		if err := pt.runPulumiCommand("pulumi-env-init", initArgs, dir, false); err != nil {
+			return err
+		}
+
+		envFile, err := func() (string, error) {
+			temp, err := os.CreateTemp("", "pulumi-env-test-*")
+			if err != nil {
+				return "", err
+			}
+			defer contract.IgnoreClose(temp)
+
+			enc := yaml.NewEncoder(temp)
+			enc.SetIndent(2)
+			if err = enc.Encode(env.Definition); err != nil {
+				return "", err
+			}
+			return temp.Name(), nil
+		}()
+		if err != nil {
+			return err
+		}
+
+		editArgs := []string{"env", "edit", name, fmt.Sprintf("--editor=cp %q", envFile)}
+		if err := pt.runPulumiCommand("pulumi-env-edit", editArgs, dir, false); err != nil {
+			return err
+		}
+	}
+
+	if len(pt.opts.Environments) != 0 {
+		envs := make([]string, len(pt.opts.Environments))
+		for i, e := range pt.opts.Environments {
+			envs[i] = pt.opts.getEnvName(e)
+		}
+
+		stackFile := filepath.Join(dir, fmt.Sprintf("Pulumi.%v.yaml", stackName))
```

Comment:
Does this work if we have steps? They all have their own folders and not sure if stack files would copy for that?

Reply:
I was wondering about that myself. I would hope it works: despite the separate directories, it seems like tests that use config just wouldn't work if stack files weren't copied. Those tests do pass, so that provided some evidence for me that this should work even for tests with steps.